examples/galaxies/plot_alhambra.py

```diff
@@ -0,0 +1,98 @@
+#
+# .. math::
+#
+#    \phi(M) = \frac{{\rm d}N}{{\rm d}M{\rm d}V} = \frac{2}{5}{\rm log}(10)\phi_*10^{\frac{2}{5}(M_*-M)(\alpha+1)}{\rm exp}\left[-10^{\frac{2}{5}(M_*-M)}\right]
+#
+# In general, :math:`\phi_*`, :math:`M_*` and :math:`\alpha` can be redshift
+# dependent, resulting in a joint redshift-magnitude distribution
+# :math:`\phi(M, z)`. In SkyPy, sampling from this joint distribution is
+# implemented in :func:`skypy.galaxy.schechter_lf` where both `phi_star` and
+# `m_star` can be either constants or functions of redshift. In this example,
+# we follow a common parameterisation where `phi_star` and `m_star` are
+# exponential and linear functions of redshift respectively. Samples are
+# generated for a given `sky_area` and over a given redshift range `z_range` up
+# to a limiting apparent magnitude `mag_lim` with shot noise. The values of the
+# parameters are taken from the B-band luminosity function for blue galaxies in
+# López-Sanjuan et al. 2017 [1]_.
+
+from astropy.cosmology import FlatLambdaCDM
+from astropy.modeling.models import Linear1D, Exponential1D
+from astropy.units import Quantity
+import numpy as np
+from skypy.galaxy import schechter_lf
+
+z_range = np.linspace(0.2, 1.0, 100)
+m_star = Linear1D(-1.03, -20.485)
+phi_star = Exponential1D(0.00312608, -43.4294)
+alpha, mag_lim = -1.29, 27
+sky_area = Quantity(1, "deg2")
+cosmology = FlatLambdaCDM(H0=70, Om0=0.3)
+redshift, magnitude = schechter_lf(z_range, m_star, phi_star, alpha,
+                                   mag_lim, sky_area, cosmology, noise=True)
+
+# %%
+# ALHAMBRA B-Band Luminosity Function
+# -----------------------------------
+#
+# Here we compare our sampled galaxy B-band magnitudes in four different
+# redshift slices to the observed B-band luminosity function from the ALHAMBRA
+# survey and the analytic luminosity function from López-Sanjuan et al. 2017.
+
+from matplotlib import pyplot as plt
+from astropy.table import Table
+
+data = Table.read("./alhambra_data.ecsv", format='ascii.ecsv')
+bins = np.linspace(-24, -12, 100)
+fig, axs = plt.subplots(nrows=2, ncols=2, constrained_layout=True)
+
+for i in range(4):
+    z_min, z_max = 0.2 * (i+1), 0.2 * (i+2)
+    mask = np.logical_and(redshift >= z_min, redshift < z_max)
+    z = np.linspace(z_min, z_max, 100)
+    dV_dz = (cosmology.differential_comoving_volume(z) * sky_area).to_value('Mpc3')
+    dV = np.trapz(dV_dz, z)
+    dM = (np.max(bins)-np.min(bins)) / (np.size(bins)-1)
+    phi = np.histogram(magnitude[mask], bins=bins)[0] / dV / dM
+    ax = axs[i//2, i%2]
+    ax.plot(bins[:-1] + dM/2, phi, label='SkyPy', color='red')
+    ax.scatter(data[f'MB{i+1}'], 10**(data[f'phi{i+1}']), marker='s', label='López-Sanjuan+17')
+    ax.text(0.05, 0.05, f"{z_min:.1f} <= z < {z_max:.1f}", transform=ax.transAxes)
+    ax.set_xlabel('M_B')
+    ax.set_ylabel('phi')
+    ax.set_yscale('log')
+    ax.set_xlim([-14, -24])
+
+axs[0,0].legend(loc=6, fontsize='small', frameon=False)
```

Comment:
Codeclimate seems to complain about this
```suggestion
axs[0, 0].legend(loc=6, fontsize='small', frameon=False)
```